### PR TITLE
[web] Smoke tests for web engine

### PR DESCRIPTION
**Pull request by nturgut**
## Description

We would like to add a simple smoke test which uses Flutter web engine and a Flutter app.

Steps before merging the PR:
- (DONE) https://github.com/flutter/flutter/pull/50835 should be merged so Flutter drive does not need selenium and java to run.
- (DONE) The chrome driver util codes should be merged either to a package or to a new repo: https://github.com/flutter/web_installers/pull/1
- PR that enables flutter drive to run the tests directly: https://github.com/flutter/flutter/pull/51084#discussion_r382279891

Then we can add a task that looks like this to Cirrus:

dart dev/webdriver_test.dart &
flutter drive --target=test_driver/scroll_perf_web.dart -d web-server --release --browser-name=chrome

## Tests

This PR only includes only test code.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: examples/hello_world/pubspec.yaml
Comment (yjbanov) [inline examples/hello_world/pubspec.yaml]: We use concrete version numbers on `test`. You can copy the current number from another `pubspec.yaml` file.

File: examples/hello_world/test_driver/smoke_web_engine.dart
Comment (yjbanov) [inline examples/hello_world/test_driver/smoke_web_engine.dart]: This and other new files need a license header.

File: examples/hello_world/test_driver/smoke_web_engine_test.dart
Comment (yjbanov) [inline examples/hello_world/test_driver/smoke_web_engine_test.dart]: This comment seems too obvious. I'd remove it.

File: examples/hello_world/pubspec.yaml
Comment (nturgut) [reply examples/hello_world/pubspec.yaml]: done.

File: examples/hello_world/test_driver/smoke_web_engine_test.dart
Comment (nturgut) [reply examples/hello_world/test_driver/smoke_web_engine_test.dart]: done!

File: examples/hello_world/test_driver/smoke_web_engine.dart
Comment (nturgut) [reply examples/hello_world/test_driver/smoke_web_engine.dart]: done

**Review by yjbanov — APPROVED** [review]

**Comment by nturgut** [comment]
Waiting for another PR to land before merging this test.

File: .cirrus.yml
Comment (yjbanov) [inline .cirrus.yml]: Should we reuse the `web_integration_tests` shard?

File: .cirrus.yml
Comment (yjbanov) [inline .cirrus.yml]: I think you'll want to `export FLUTTER_WEB=true` before running any of the web commands. Otherwise the tool will refuse to do any web stuff.

**Review by yjbanov — APPROVED** [review]

File: .cirrus.yml
Comment (nturgut) [reply .cirrus.yml]: It is very different as the content goes. Especially with the template. May be I am missing something. What is the advantage if we use it? We should also change/remove the template from that shard then.

File: .cirrus.yml
Comment (nturgut) [reply .cirrus.yml]: the run is temporary. so this will also be gone. we won't need it one the PR is in: https://github.com/flutter/flutter/pull/51084 (I won't merge my PR before that one)

Comment (nturgut) [reply .cirrus.yml]: Now PR #51084 is in, no need for flutter run!

File: .cirrus.yml
Comment (yjbanov) [reply .cirrus.yml]: A smoke test is an integration test. It's just a simple integration test. So I thought it belonged in the integration test shard. We also probably don't want too many shards as they all run simultaneously and occupy machines. I imagine eventually we'll have a hundred integration tests, and we'll shard them into N shards as we see fit, similarly to what we do with widget tests. This way we don't have hand-maintained specialized shards.

File: examples/hello_world/lib/main.dart
Comment (jonahwilliams) [inline examples/hello_world/lib/main.dart]: Could you open a new PR to revert this change?

Comment (yjbanov) [reply examples/hello_world/lib/main.dart]: It adds a `key` to the `Text` widget, which is needed by the test.

Comment (nturgut) [reply examples/hello_world/lib/main.dart]: Why are we reverting the change? The test are passing. I checked the smoke test after the PR went in. It always passed.

**Pull request merged**
